cpu/stm32_common/periph/gpio.c

```diff
@@ -253,6 +257,30 @@ void isr_exti(void)
 }
 #endif /* MODULE_PERIPH_GPIO_IRQ */
 
-#else
+#ifdef MODULE_PERIPH_GPIO_ABC
+static inline __attribute__((always_inline)) void _delay(int32_t loops)
+{
+    __asm__ volatile(
+        "0:" "SUBS %[iterator], #1;"
+        "BPL 0b;"
+        : [iterator]"+r"(loops)
+    );
+}
+
+void gpio_set_for(gpio_t pin, int delay)
+{
+    _port(pin)->BSRR = (1 << _pin_num(pin));
+    _delay(delay);
+}
+
+void gpio_clear_for(gpio_t pin, int delay)
+{
+    _port(pin)->BSRR = (1 << (_pin_num(pin) + 16));
+    _delay(delay);
+}
+
```

Comment:
Maybe a better idea would be to allow for `gpio_set()` to be inlinable so everyone benefits from the reduced overhead.

Then you could have a generic `gpio_set_for()`/`gpio_clear_for()` wrapper.

Reply:
I'm not that sure about this. If the GPIO API remains a periph-only API, that would surely be an option. (I'm however not sure if it would increase the ROM size for non-LTO builds in cases where the value of the `pin` parameter is not known at compile time.)

I do hope however that the GPIO API opens up for external GPIOs as well. @gschorcht  is currently working on that. I would like to wait for that to happen before checking if implementing the GPIO API as `static inline` functions is generally feasible.